libass/ass_fontconfig.c

```diff
@@ -109,23 +110,26 @@ static void scan_fonts(FcConfig *config, ASS_FontProvider *provider)
         // simple types
         result  = FcPatternGetInteger(pat, FC_SLANT, 0, &meta.slant);
         result |= FcPatternGetInteger(pat, FC_WIDTH, 0, &meta.width);
-        result |= FcPatternGetInteger(pat, FC_WEIGHT, 0, &weight);
+        result |= FcPatternGetDouble(pat, FC_WEIGHT, 0, &weight);
         result |= FcPatternGetInteger(pat, FC_INDEX, 0, &index);
         if (result != FcResultMatch)
             continue;
 
         // fontconfig uses its own weight scale, apparently derived
         // from typographical weight. we're using truetype weights, so
-        // convert appropriately
-#if FC_VERSION >= 21191
-        meta.weight = FcWeightToOpenType(weight);
+        // convert appropriately.
+#if FC_VERSION >= 21292
+        meta.weight = FcWeightToOpenTypeDouble(weight) + 0.5;
+#elif FC_VERSION >= 21191
+        // Versions prior to 2.12.92 only had integer precision.
+        meta.weight = FcWeightToOpenType(weight + 0.5) + 0.5;
 #else
         // On older fontconfig, FcWeightToOpenType is unavailable, and its inverse was
         // implemented more simply, using an if/else ladder instead of linear interpolation.
         // We implement an inverse of that ladder here.
         // We don't expect actual FC caches from these versions to have intermediate
         // values, so the average checks are only for completeness.
-#define AVG(x, y) ((x + y) / 2)
```

Comment:
I’d have written this as `(x + y) * 0.5`. But it makes no difference given that we only ever call this with small integer constant expressions.